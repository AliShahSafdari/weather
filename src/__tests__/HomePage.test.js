import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';
import axios from 'axios';
import ShowAll from '../components/HomePage';

jest.mock('axios');

const arr = [
  {
    coord: { lon: 69.4167, lat: 34.5 },
    weather: [{
      id: 800, main: 'Clear', description: 'clear sky', icon: '01d',
    }],
    base: 'stations',
    main: {
      temp: 294.11,
      feels_like: 293.07,
      temp_min: 294.11,
      temp_max: 294.11,
      pressure: 1008,
      humidity: 31,
      sea_level: 1008,
      grnd_level: 811,
    },
    visibility: 10000,
    wind: { speed: 1.33, deg: 303, gust: 1.51 },
    clouds: { all: 6 },
    dt: 1688000991,
    sys: { country: 'AF', sunrise: 1687997534, sunset: 1688049532 },
    timezone: 16200,
    id: 1138957,
    name: 'Kabul',
    cod: 200,
  },
  {
    coord: { lon: 68.4269, lat: 33.5536 },
    weather: [{
      id: 800, main: 'Clear', description: 'clear sky', icon: '01d',
    }],
    base: 'stations',
    main: {
      temp: 293.71,
      feels_like: 293.13,
      temp_min: 293.71,
      temp_max: 293.71,
      pressure: 1008,
      humidity: 50,
      sea_level: 1008,
      grnd_level: 786,
    },
    visibility: 10000,
    wind: { speed: 1.54, deg: 352, gust: 1.75 },
    clouds: { all: 0 },
    dt: 1688001287,
    sys: { country: 'AF', sunrise: 1687997924, sunset: 1688049617 },
    timezone: 16200,
    id: 1141269,
    name: 'Ghazni',
    cod: 200,
  },
];

const reducer = (state = {
  weathers: { weatherData: arr },
}) => state;

const store = configureStore({ reducer });

describe('should test My home page', () => {
  it('should test the snapshot of home page', async () => {
    await axios.get.mockResolvedValue({ weather: arr });
    const homePage = renderer.create(
      <MemoryRouter>
        <Provider store={store}>
          <ShowAll />
        </Provider>
      </MemoryRouter>,
    ).toJSON();
    expect(homePage).toMatchSnapshot();
  });
});
