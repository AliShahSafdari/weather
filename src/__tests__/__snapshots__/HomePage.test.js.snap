// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test My home page should test the snapshot of home page 1`] = `
<div>
  <header
    className="header"
  >
    <h1
      className="header-title"
    >
      WeatherHub: Explore Weather in Major Cities 
    </h1>
    <div
      className="header-content"
    >
      <div
        className="header-filter"
      >
        <span>
          Filter
        </span>
        <select
          onChange={[Function]}
          value="All"
        >
          <option
            value="All"
          >
            All
          </option>
          <option
            value="Hot"
          >
            Hot
          </option>
          <option
            value="Temperate"
          >
            Temperate
          </option>
          <option
            value="Cold"
          >
            Cold
          </option>
        </select>
      </div>
      <h2>
        Most views
      </h2>
      <div
        className="icons"
      >
        <img
          alt="microphone"
          className="icons-img"
          src="green-radio-microphone-14640.png"
        />
        <img
          alt="setting"
          className="icons-img"
          src="settings-5666.png"
        />
      </div>
    </div>
  </header>
  <div
    className="container"
  >
    <div
      className="city"
    >
      <a
        className="link"
        href="/details/1138957"
        onClick={[Function]}
      >
        <div
          className="city-image"
        >
          <img
            alt="weatherIcon"
            className="city-image-one"
            src="yellow-sun-and-blue-cloud-16528.png"
          />
          <img
            alt="go on "
            src="right-arrow-6421.png"
          />
        </div>
        <h3>
          Kabul
        </h3>
        <p>
          20.96
           
          °C
        </p>
        <p>
           
          Humidity:
           
          31
          %
        </p>
      </a>
    </div>
    <div
      className="city"
    >
      <a
        className="link"
        href="/details/1141269"
        onClick={[Function]}
      >
        <div
          className="city-image"
        >
          <img
            alt="weatherIcon"
            className="city-image-one"
            src="yellow-sun-and-blue-cloud-16528.png"
          />
          <img
            alt="go on "
            src="right-arrow-6421.png"
          />
        </div>
        <h3>
          Ghazni
        </h3>
        <p>
          20.56
           
          °C
        </p>
        <p>
           
          Humidity:
           
          50
          %
        </p>
      </a>
    </div>
  </div>
</div>
`;
